pynxtools/dataconverter/helpers.py

```diff
@@ -529,6 +545,11 @@ def try_undocumented(data, nxdl_root: ET.Element):
         nxdl_path = convert_data_converter_dict_to_nxdl_path(path)
 
         if entry_name == "@units":
+            if (
+                path.rsplit("/", 1)[0] in data.get_documented()
+                and path in data.undocumented
+            ):
+                del data.undocumented[path]
```

Comment:
I don't want to dwell too much on this. And we can merge this in if it doesn't break anything. But this looks a bit odd to me.

We are trying to remove dangling 'field/@units' from data.undocumented where there is a 'field' path available in documented. This should already be done below in line 564. On top of that, do you really just want to remove the units without placing them in their respective optionality? Do your nexus files still write the units being removed here?

I don't want us to deal with this right now. But it will be nice to have a little reflection to not get confused later.

Reply:
> We are trying to remove dangling 'field/https://github.com/Units' from data.undocumented where there is a 'field' path available in documented. This should already be done below in line 564. 

In this case it is not, because only the `@units` are in undocumented and therefore we never check the associated `@units`. I don't know why this happens, but I think it might be due to that we only inherit the units and don't state them explicitly.

> On top of that, do you really just want to remove the units without placing them in their respective optionality? Do your nexus files still write the units being removed here?

Good point! Probably not. I will check.

> I don't want us to deal with this right now. But it will be nice to have a little reflection to not get confused later.

No, that's perfect! That's why I wanted you to review this parts. So I don't make anything stupid here 🙈

Reply:
It's fixed. `@units` are now written again.